piet/src/text.rs

```diff
@@ -94,6 +135,26 @@ pub trait TextLayout {
     fn hit_test_text_position(&self, text_position: usize) -> Option<HitTestTextPosition>;
 }
 
+/// Metadata about each line in a text layout.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct LineMetric {
+    /// Index (in code units) of the start of the line, offset from the beginning of the text.
+    pub line_start_offset: usize,
+
+    /// Line length (in code units), but offset from the beginning of the text. So it's the length
+    /// of this line summed with the lengths of all the lines before it.
+    ///
+    /// Does not include trailing whitespace.
+    pub line_end_offset: usize,
+
+    /// Line width.
```

Comment:
What are the units? I would expect this to be in pixels or something but it's not clear.

I'm also unsure about this name, 'cumulative' implies that this includes the widths of preceding lines or something? If that's the case we  should document it, but otherwise this can just be `width`?

Reply:
Hmmm... this was in the last PR for line breaking, so I carried it over. But looking at it again I'm not sure exactly why we need cumulative width. Is it better to just use width?

Reply:
I think we want just "width", in pixels. Also I think this comment was maybe in response to this?


> Raph: I think it makes sense not to do cumulative width of lines, but it does make sense to do cumulative code unit counts. This is not a tricky thing, a "cumulative code unit count" is just an offset.




https://github.com/linebender/piet/pull/134#issuecomment-595604575